### PR TITLE
Feature/workflow manager

### PR DESCRIPTION
**Pull request by reisingerf**

### CODE REVIEW

File: lib/workload/stateless/stacks/workflow-manager/workflow_manager/tests/factories.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    workflow_ref_id =  str(uuid.uuid4())
+    execution_engine =  "ICAv2"
+    approval_state =  "NATA"
+
+
+class PayloadFactory(factory.django.DjangoModelFactory):
+    class Meta:
+        model = Payload
+
+    payload_type =  "success_payload"
```
Comment (victorskl) [inline lib/workload/stateless/stacks/workflow-manager/workflow_manager/tests/factories.py:39]: `payload_type` is obsolete. Need add `version` as replacement.

File: lib/workload/stateless/stacks/workflow-manager/workflow_manager/tests/factories.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+
+class WorkflowFactory(factory.django.DjangoModelFactory):
+    class Meta:
+        model = Workflow
+
+    workflow_name =  "TestWorkflow"
+    workflow_version =  "1.0"
+    workflow_ref_id =  str(uuid.uuid4())
```
Comment (victorskl) [inline lib/workload/stateless/stacks/workflow-manager/workflow_manager/tests/factories.py:30]: `workflow_ref_id` is obsolete. Need add `execution_engine_pipeline_id` as replacement.

File: lib/workload/stateless/stacks/workflow-manager/workflow_manager/tests/factories.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+class WorkflowRunFactory(factory.django.DjangoModelFactory):
+    class Meta:
+        model = WorkflowRun
+
+    _uid = str(uuid.uuid4())
+    portal_run_id =  f"20240130{_uid[:8]}"
+    execution_id =  _uid
+    workflow_run_name =  f"TestWorkflowRun{_uid[:8]}"
+    status =  "ready"
```
Comment (victorskl) [inline lib/workload/stateless/stacks/workflow-manager/workflow_manager/tests/factories.py:52]: Cap pls. `status = "READY"`

File: lib/workload/stateless/stacks/workflow-manager/workflow_manager/tests/factories.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+from workflow_manager.models.workflow_run import WorkflowRun
+from workflow_manager.models.payload import Payload
+
+
+class  TestConstant(Enum):
+    workflow_name = "TestWorkflow1"
+    payload = {
+        "key": "value",
+        "foo": uuid.uuid4(),
+        "bar": datetime.now().astimezone(ZoneInfo('Australia/Sydney')),
```
Comment (victorskl) [inline lib/workload/stateless/stacks/workflow-manager/workflow_manager/tests/factories.py:19]: This zone info should work, right? Perhaps, we should run `python manage.py generate_mock_data` locally to confirm it.

**Review by victorskl — APPROVED** [review]
LGTM. Just need to attend comments in test stub factories.

File: lib/workload/stateless/stacks/workflow-manager/workflow_manager/tests/factories.py (hunk 1 above)
Comment (reisingerf) [reply lib/workload/stateless/stacks/workflow-manager/workflow_manager/tests/factories.py]: Fixed with [5afd11a](https://github.com/umccr/orcabus/pull/325/commits/5afd11af1e5b14e4ef0feefb3618fbfff5fe35f6)

File: lib/workload/stateless/stacks/workflow-manager/workflow_manager/tests/factories.py (hunk 2 above)
Comment (reisingerf) [reply lib/workload/stateless/stacks/workflow-manager/workflow_manager/tests/factories.py]: Fixed with [5afd11a](https://github.com/umccr/orcabus/pull/325/commits/5afd11af1e5b14e4ef0feefb3618fbfff5fe35f6)

File: lib/workload/stateless/stacks/workflow-manager/workflow_manager/tests/factories.py (hunk 3 above)
Comment (reisingerf) [reply lib/workload/stateless/stacks/workflow-manager/workflow_manager/tests/factories.py]: Fixed with [5afd11a](https://github.com/umccr/orcabus/pull/325/commits/5afd11af1e5b14e4ef0feefb3618fbfff5fe35f6)

File: lib/workload/stateless/stacks/workflow-manager/workflow_manager/tests/factories.py (hunk 4 above)
Comment (reisingerf) [reply lib/workload/stateless/stacks/workflow-manager/workflow_manager/tests/factories.py:19]: Yes, that should work. 
Example mock generation work with [5afd11a](https://github.com/umccr/orcabus/pull/325/commits/5afd11af1e5b14e4ef0feefb3618fbfff5fe35f6)

**Pull request merged**